app/views/appointments/index.html.erb

```diff
@@ -16,13 +16,23 @@
       <% @appointments.each do |appointment| %>
         <tr>
           <td><i class="fa-regular fa-calendar"></i></td>
-          <td><%=appointment.title%></td>
-          <td> <%=appointment.start_time.to_fs(:short) %></td>
+          <td><%= appointment.title %></td>
+          <td><%= appointment.start_time.to_fs(:short) %></td>
           <td><%= link_to "Muestra esta cita", appointment,class:"btn btn-outline-secondary btn-sm" %></td>
+          <td><%= link_to "Posponer", new_appointment_path(preloaded_appointment_id: appointment.id), class: "btn btn-outline-secondary btn-sm posponer", id: "appointment_#{appointment.id}"%></td>
         </tr>
       <% end %>
     </tbody>
   </table>
 </div>
+<script>
+  $(document).ready(function() {
+    $('.posponer').click(function(event) {
+      event.preventDefault();
+      var appointmentId = $(this).attr('id').replace('appointment_', '');
+      window.location.href = "/appointments/new?preloaded_appointment_id=" + appointmentId;
+    });
+  });
```

Comment:
I don't think this script is needed, it is sending you to appointments#new view with a parameter but line 22 is also sending you there. Try removing this script and test your link.

Reply:
I was testing my link and it worked but I realized that if I need the variable to put because I have for each patient their link, then postpone based on their id, specific works
<img width="1169" alt="Captura de pantalla 2024-06-04 a la(s) 7 20 29 p m" src="https://github.com/NadiaArroyo25/prueba1/assets/130617270/b25a3806-3314-44d9-b9bf-0d28cfd191f8">

Reply:
You are including the appointment ID for each link in your regular link
```ruby
new_appointment_path(preloaded_appointment_id: appointment.id)
```
So, you don't need this script.

Reply:
D: I'm sorry, I hadn't understood, and I was clinging to the idea that the jquery of that script had to be taken care of